File: Example/Tests/Objective-C/NyanCat.m
//
//  NyanCat.m
//  SuperMessagingProxyTests
//
//  Created by Aleksandr Gusev on 20/05/16.
//  Copyright © 2016 Alexander Gusev. All rights reserved.
//

#import "NyanCat.h"

NS_ASSUME_NONNULL_BEGIN

@implementation NyanCat

@synthesize awesomenessLevel = _awesomenessLevel;

- (instancetype)init {
    if (self = [super init]) {
        _awesomenessLevel = 10;
    }
    return self;
}

- (NSString *)exclamation {
    return @"Nyan!";
}

- (NSString *)says {
    return @"Nyan";
}

- (CatDescriptor)descriptor {
    CatDescriptor descriptor;
    memset(&descriptor, 0, sizeof(descriptor));
    strncpy(descriptor.name, "NyanCat", sizeof("NyanCat"));
    return descriptor;
}

+ (NSString *)classSays {
    return @"Class Nyan";
}

+ (NSInteger)classAwesomenessLevel {
    return 9;
}

@end

NS_ASSUME_NONNULL_END

File: Example/Tests/Swift/NyanCat.swift
//
//  NyanCat.swift
//  SGVSuperMessagingProxy
//
//  Created by Aleksandr Gusev on 19/05/16.
//  Copyright © 2016 Alexander Gusev. All rights reserved.
//

class NyanCat: Cat {
    override func says() -> String {
        return "Nyan"
    }
    
    override var exclamation: String {
        return "Nyan!"
    }
    
    private var _awesomenessLevel: Int = 10
    
    override var awesomenessLevel: Int {
        get {
            return _awesomenessLevel
        }
        set {
            _awesomenessLevel = newValue
        }
    }
    
    override class func classSays() -> String {
        return "Class Nyan"
    }
    
    override class var classAwesomenessLevel: Int {
        return 10
    }
}

